### DISCUSSION TITLE
自定义UDAF函数，类中的各个方法在什么情况下才会调用？

### DISCUSSION THREAD

**Discussion by zhangjjzyl** [apache/doris#33355] [A - General / Q&A]
>doris版本：2.0.4

`create add getValue destroy `方法每次查询时都会调用，但是`serialize deserialize merge` 则不一定会触发，请问各个方法的调用机制是什么？什么情况才会触发调用？

**Comment by yagagagaga** [comment]
You can take a look at the description of this [interface](https://github.com/apache/doris/blob/master/be/src/vec/aggregate_functions/aggregate_function.h).
